dependencies.R

```diff
@@ -11,7 +11,33 @@ github_packages <- c(
 lapply(github_packages, remotes::install_github)
 
 # CRAN
-install.packages("kableExtra")
+options(repos = c("https://cloud.r-project.org/"))
+ncores <- parallel::detectCores(all.tests = FALSE, logical = TRUE)
+if (!require("kableExtra")) install.packages("kableExtra", upgrade = "never", Ncpus = ncores)
+if (!require("tinytex")) install.packages("tinytex", upgrade = "never", Ncpus = ncores)
 
-# Install missing fonts
-system("tlmgr install courier ec")
+# Conditionally install TinyTex
+if(!dir.exists(paste(Sys.getenv("RUNNER_TEMP"), "TinyTeX", sep="/"))) {
+# nolint start
+  tinytex_installer <- '
+export TINYTEX_DIR=${RUNNER_TEMP}/TinyTeX
+wget -qO- "https://raw.githubusercontent.com/yihui/tinytex/master/tools/install-unx.sh" | sh -s - --admin --no-path
+mkdir -p ${RUNNER_TEMP}/TinyTeX
+cp -r ~/.TinyTeX/. ${RUNNER_TEMP}/TinyTeX
+rm -rf ~/.TinyTeX
+${RUNNER_TEMP}/TinyTeX/bin/*/tlmgr path add
+tlmgr install latex-bin luatex xetex ae bibtex context inconsolata listings makeindex metafont mfware parskip pdfcrop tex tools url xkeyval courier ec
```

Comment:
Ideally I'd like this to be an input to the action - i.e. add additional TeX packages if they are missing or needed by a package. 
Let's keep these packages as defaults, but let's also provide the ability to add extra pacakges.

Reply:
the whole list without `courier` and `ec` comes from rocker/verse - they are installed by default.
I added parameter that allows install more of them then only those two.